test/tools/unified-spec-runner/match.ts

```diff
@@ -523,6 +567,51 @@ export function matchesEvents(
   }
 }
 
+export function compareLogs(
+  expected: ExpectedLogMessage[],
+  actual: ExpectedLogMessage[],
+  entities: EntitiesMap
+): void {
+  expect(actual).to.have.lengthOf(expected.length);
+
+  for (const [index, actualLog] of actual.entries()) {
+    const rootPrefix = `expectLogMessages[${index}]`;
+    const expectedLog = expected[index];
+
+    // Check that log levels match
+    expect(actualLog).to.have.deep.property('level', expectedLog.level);
+
+    // Check that components match
+    expect(actualLog).to.have.deep.property('component', expectedLog.component);
+
+    // NOTE: The spec states that if the failureIsRedacted flag is present, we
+    // must assert that a failure occurred.
+    if (expectedLog.failureIsRedacted !== undefined) {
```

Comment:
you may as well assert that `expectedLog.failureIsRedacted` is a boolean in the line below, we might not be running the `invalid` tests, but there's no reason not to have the guard